pages/api/auth/register.js

```diff
@@ -0,0 +1,37 @@
+import bcrypt from "bcrypt";
+import sql from "~/lib/postgres";
+import * as yup from "yup";
+
+const schema = yup.object().shape({
+  email: yup.string().email().required().lowercase(),
+  hash_password: yup.string().required().min(6),
+});
+
+const handler = async ({ method, body: { email, password } }, res) => {
+  if (method === "POST") {
+    const user = {
+      email: email,
+      hash_password: password,
+      is_admin: false,
+    };
+    const userValid = await schema.validate(user);
+    if (userValid) {
+      const userExists = await sql`SELECT * FROM users WHERE email=${email}`;
+      userValid.hash_password = await bcrypt.hash(password, 10);
+      if (userExists.count > 0) {
+        return res
+          .status(400)
+          .json({ message: "There is already an account with this email" });
+      } else {
+        const [newUser] = await sql`INSERT INTO users ${sql(
+          userValid,
+          ...Object.keys(userValid)
+        )} RETURNING *;`;
+        return res.json({ users: newUser });
```

Comment:
Tiny thing but may trip you up later on, might be better named as `user` as opposed to `users` as I presume this will always be a single user who has made an account.